### ISSUE TITLE
Owncast v0.0.13 keeps crashing when streamed to by StreamYard

### ISSUE THREAD

**Issue by KyleMaas** [owncast/owncast#2943]
### Share your bug report, feature request, or comment.

When streaming to Owncast from StreamYard, previously everything worked perfectly.  But then about two months ago, it started crashing Owncast about every 90-180 seconds or so (the runtime isn't consistent).  Owncast immediately restarts - I can see this in the logs - but it stops the stream for any viewers watching it.  This sequence will repeat for as long as I'm streaming, dumping off all the viewers every few minutes.  No further debugging information is available in the logs, no matter what detail level I log at.  Clients will be watching just fine, and then they start getting errors, and then about 4 seconds later the logs show Owncast's startup sequence.

Any idea what might be causing this?

OS: Rocky Linux 9
Owncast version: v0.0.13

**Comment by KyleMaas** [comment]
I should note: I've waited this long because I've been trying to rule out user error on my end.  I tried completely rebuilding the server.  Tried switching from Ubuntu to Rocky Linux.  Have tried completely wiping the database and reloading Owncast from scratch.  None of it has worked.  But I did want to rule out as many possibilities from my end as I could.

**Comment by gabek** [comment]
Just for some high level clarification, when you say "crash" do you mean the server process exits unexpectedly, or are you saying the stream becomes unstable for viewers, but Owncast itself is still running?

**Comment by KyleMaas** [comment]
@gabek The server process exits, without any log messages, and then restarts again.  Presumably systemd is triggering the restart when it sees the process die but I can't tell for sure.

**Comment by gabek** [comment]
Ok cool. In that case you should be able to pull crash logs from systemd to see what actually happened. Owncast itself won't have logs from the crash, but your system will.

An alternative is to run Owncast manually (outside of systemd) and monitor your console to see exactly what's going on when it exits.

Thanks for your help, the crash log should help me track down the issue.

**Comment by KyleMaas** [comment]
@gabek Ah, you're right.  systemd did have the logs:

```
Apr 24 16:02:37 [hostname redacted] owncast[745]: panic: runtime error: invalid memory address or nil pointer dereference
Apr 24 16:02:37 [hostname redacted] owncast[745]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa0ab67]
Apr 24 16:02:37 [hostname redacted] owncast[745]: goroutine 696 [running]:
Apr 24 16:02:37 [hostname redacted] owncast[745]: github.com/owncast/owncast/notifications/browser.(*Browser).Send(0xc0007eaa50, {0xc0000f0280, 0x26d}, {0xc000260568?, 0x0?}, {0xc000260680?, 0x0?})
Apr 24 16:02:37 [hostname redacted] owncast[745]:         /go/src/github.com/owncast/owncast/notifications/browser/browser.go:75 +0x247
Apr 24 16:02:37 [hostname redacted] owncast[745]: github.com/owncast/owncast/notifications.(*Notifier).notifyBrowserPush(0xc0001940c0)
Apr 24 16:02:37 [hostname redacted] owncast[745]:         /go/src/github.com/owncast/owncast/notifications/notifications.go:105 +0x146
Apr 24 16:02:37 [hostname redacted] owncast[745]: github.com/owncast/owncast/notifications.(*Notifier).Notify(0xc0001940c0)
Apr 24 16:02:37 [hostname redacted] owncast[745]:         /go/src/github.com/owncast/owncast/notifications/notifications.go:183 +0x27
Apr 24 16:02:37 [hostname redacted] owncast[745]: github.com/owncast/owncast/core.startLiveStreamNotificationsTimer.func1({0x2406390, 0xc0005dd0c0})
Apr 24 16:02:37 [hostname redacted] owncast[745]:         /go/src/github.com/owncast/owncast/core/streamState.go:188 +0x1c7
Apr 24 16:02:37 [hostname redacted] owncast[745]: created by github.com/owncast/owncast/core.startLiveStreamNotificationsTimer
Apr 24 16:02:37 [hostname redacted] owncast[745]:         /go/src/github.com/owncast/owncast/core/streamState.go:169 +0xaa
```

**Comment by gabek** [comment]
Awesome, thanks so much!

**Comment by KyleMaas** [comment]
No problem!  Thanks for helping me find it!  I've been stuck on this for a long time now thinking it might be something I was doing wrong.

**Comment by KyleMaas** [comment]
If you have any idea of what's causing this, is there any kind of a workaround I could put into place to patch this up until v0.1.0 stabilizes and is published?

**Comment by gabek** [comment]
From that log it looks like a bug with browser push notifications. Maybe disabling that feature could work around it? Or maybe you've already disabled it and that's causing the bug? I'll be looking at it soon and I should be able to track down the actual issue and give you a better answer.

**Comment by KyleMaas** [comment]
Awesome!  Thank you so much for the help.

**Comment by KyleMaas** [comment]
@gabek Turning off notifications seems to have alleviated the problem.  Thanks!

**Comment by gabek** [comment]
How quickly I forget. This was already fixed when it was reported in #2855. Sorry about that!

**Comment by KyleMaas** [comment]
Even better!  That means it's already fixed.  Thanks!
